src/compiler/transformer.ts

```diff
@@ -288,12 +289,7 @@ namespace ts {
         function emitNodeWithNotification(hint: EmitHint, node: Node, emitCallback: (hint: EmitHint, node: Node) => void) {
             Debug.assert(state < TransformationState.Disposed, "Cannot invoke TransformationResult callbacks after the result is disposed.");
             if (node) {
-                if (isEmitNotificationEnabled(node)) {
-                    onEmitNode(hint, node, emitCallback);
-                }
-                else {
-                    emitCallback(hint, node);
-                }
+                onEmitNode(hint, node, emitCallback);

```

Comment:
There is a negative aspect of moving this to `TransformationResult` and `PrintHandlers`, in that any existing API consumers *aren't* doing this, so we could potentially break them by removing this check. We should err on the side of caution and leave this check here (and merely use the new `PrintHandlers` hook as an optimization).